src/main/java/codeu/model/store/basic/MentionStore.java

```diff
@@ -42,6 +43,11 @@ public static MentionStore getTestInstance(PersistentStorageAgent persistentStor
 
   /** The in-memory list of Mentions. */
   private List<Mention> mentions;
+  
+  /** Access the current set of mentions known to the application. */
+  public List<Mention> getAllMentions() {
+    return mentions;
```

Comment:
So this is the scary thing that you should be aware of.  The member variable "mentions" is effectively a global in this app. When you return it, you don't actually have any guarantee that the calling code won't do some dumb thing like List<Mention> mentions = mentionStore.getMentions(); .... a bunch of stuff .... mentions.clear();
This would delete all mentions everywhere!  "But why would someone do that?" you ask -- well, it can be as simple as a bug, or a mistaken impression about how java garbage collection works, or it can just be badly written code. Also, you sort the list in place in the servlet -- so in a sense you're already modifying the list.

I would instead return an ImmutableList, and make it a copy of the original list, the way getAllUsers works.